pkg/util/types/tags.go

```diff
@@ -64,14 +64,17 @@ func (t ELBv2Tags) ServiceNameAndPort() (name string, port intstr.IntOrString, e
 		} else {
 			name = p[1]
 		}
-	} else {
+	}
+
+	if name == "" {
 		return "", intstr.IntOrString{}, fmt.Errorf("kubernetes.io/service-name tag is missing")
 	}
 
 	if v, ok := t.Get("kubernetes.io/service-port"); ok {
 		port = intstr.Parse(v)
 	} else {
-		return "", intstr.IntOrString{}, fmt.Errorf("kubernetes.io/service-port is missing")
+		port = intstr.Parse("1")
```

Comment:
will this be an issue? i think it's better to keep the original one..

Reply:
It was necessary to upgrade from an old version of the controller that didnt have the `kubernetes.io/service-port` tag yet. TBH now that we are using TG port 1, i need to look and see if this tag is still useful

Reply:
the service-port is what as specified in ingress spec, e.g.(80/http), it's not same as the target port(port on pods) 😄

Reply:
so the tag is still useful but if we come across a TG w/o it, we shouldn't abort the assembly by returning an error here. perhaps a `intstr.Parse("0")` to identify it as an invalid TG and force a recreation of it?

Reply:
i think it’s fine to use port0, since the worse case is the targetgroup will be orphaned :smile: